(window.webpackJsonp=window.webpackJsonp||[]).push([[19],{452:function(r,t,a){"use strict";a.r(t);var _=a(34),e=Object(_.a)({},(function(){var r=this,t=r.$createElement,a=r._self._c||t;return a("ContentSlotsDistributor",{attrs:{"slot-key":r.$parent.slotKey}},[a("h2",{attrs:{id:"_1-异常继承图"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#_1-异常继承图"}},[r._v("#")]),r._v(" 1. 异常继承图")]),r._v(" "),a("p",[a("img",{attrs:{src:"https://tva1.sinaimg.cn/large/00831rSTgy1gctn9v9q22j31om0si1kx.jpg",alt:"异常类关系图"}})]),r._v(" "),a("h2",{attrs:{id:"_2-error-异常"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#_2-error-异常"}},[r._v("#")]),r._v(" 2. Error 异常")]),r._v(" "),a("p",[r._v("error 是系统错误,不能进行捕捉,不能通过程序来进行错误处理. 常见的比如 OutOfMemoryError 之类，都是 Error 的子类。")]),r._v(" "),a("h2",{attrs:{id:"_3-exception-异常"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#_3-exception-异常"}},[r._v("#")]),r._v(" 3. Exception 异常")]),r._v(" "),a("p",[r._v("又叫可捕捉异常,其中又分为 运行时异常和编译时异常"),a("br"),r._v("\n可以 try,catch 进行捕捉,或者通过程序优化来避免.")]),r._v(" "),a("ul",[a("li",[a("strong",[r._v("运行时异常")]),r._v(": RuntimeException")]),r._v(" "),a("li",[a("strong",[r._v("编译时异常")]),r._v(": 在源代码编译期间必须进行处理")])]),r._v(" "),a("h2",{attrs:{id:"_4-try-catch-finally-的执行过程-如果存在-return-语句"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#_4-try-catch-finally-的执行过程-如果存在-return-语句"}},[r._v("#")]),r._v(" 4. try-catch-finally 的执行过程？如果存在 return 语句")]),r._v(" "),a("ul",[a("li",[r._v("Try:\n"),a("ul",[a("li",[r._v("try 里面可以进行 return")]),r._v(" "),a("li",[r._v("try 里面的语句出错了可以被 catch 捕获到")])])]),r._v(" "),a("li",[r._v("Catch:\n"),a("ul",[a("li",[r._v("try 捕获到异常后悔进入这个方法块")]),r._v(" "),a("li",[r._v("这个里面也可以进行 return")])])]),r._v(" "),a("li",[r._v("Fianlly:\n"),a("ul",[a("li",[r._v("无论如何这个里面的方法都会执行")]),r._v(" "),a("li",[a("code",[r._v("在 return 前,Catch 之后执行")])]),r._v(" "),a("li",[a("code",[r._v("可以进行return")]),r._v(" .如果在 fianlly 中写了 return,这个方法返回的结果永远都是 "),a("code",[r._v("fianlly 中返回的结果")])])])])])])}),[],!1,null,null,null);t.default=e.exports}}]);